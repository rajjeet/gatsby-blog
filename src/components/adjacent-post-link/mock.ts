export const previous = {
  fields: { slug: '/previous/' },
  frontmatter: { title: 'Introducing the New React Concurrent Mode' },
};
export const next = {
  fields: { slug: '/next/' },
  frontmatter: { title: 'The Implications of React Concurrent Mode' },
};
