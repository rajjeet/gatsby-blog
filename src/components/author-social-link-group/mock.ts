import { TProps } from './types';

export const makeProps = (): TProps => ({
  className: '',
});

