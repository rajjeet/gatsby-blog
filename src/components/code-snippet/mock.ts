import { TProps } from './types';

export const makeProps = (): TProps => ({
  children: `
const mainReducer = (state, action) => {
  console.log('Before', state);
  console.log('Action', action);
  let result = {
    count: countReducer(state.count, action)      
  };
  console.log('After', result);
  return result
};      
`,
  fileName: 'index.js',
});
