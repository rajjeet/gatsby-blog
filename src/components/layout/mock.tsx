import React from 'react';
import { TProps } from './types';
import { createMockGatsbyImageSharpFluid } from '../../utils/testing';
import { siteMetadata } from '../../../gatsby-config';

export const makeProps = (): TProps => ({
  children: <div>Sample child</div>,
  data: {
    file: createMockGatsbyImageSharpFluid.file,
    site: siteMetadata,
  },
});

