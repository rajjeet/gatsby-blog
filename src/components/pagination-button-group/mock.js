export const makeProps = () => ({
  className: {}, currentPage: 1, numOfPages: 5, paginationSlug: '',
});
