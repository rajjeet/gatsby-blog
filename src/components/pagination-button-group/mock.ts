export const makeProps = ({ currentPage = 1, numOfPages = 5 } = {}): TProps => ({
  className: '', currentPage, numOfPages, paginationSlug: '',
});

export type TProps = {
  className: string;
  currentPage: number;
  numOfPages: number;
  paginationSlug: string;
};
