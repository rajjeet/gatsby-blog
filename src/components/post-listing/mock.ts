import { TProps } from './types';
import { mockPosts } from '../../fixtures/posts';

export const makeProps = ({
  posts = mockPosts,
  heading = 'Post Listing',
  numOfPages = 5,
  currentPage = 1,
  paginationSlug = '/sample_slug/',
} = {}): TProps => ({
  posts,
  heading,
  numOfPages,
  currentPage,
  paginationSlug,
});

