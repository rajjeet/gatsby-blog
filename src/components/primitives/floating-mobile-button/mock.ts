import { faTimes } from '@fortawesome/free-solid-svg-icons';

export const makeProps = () => ({
  icon: faTimes,
});
