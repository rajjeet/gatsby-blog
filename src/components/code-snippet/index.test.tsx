import React from 'react';
import { render } from '@testing-library/react';
import { CodeSnippet } from './index';
import { makeProps } from './mock';

describe('<CodeSnippet />', () => {
  it('should render', () => {
    const { asFragment } = render(<CodeSnippet {...makeProps()} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should have a Copy button', () => {
    const { getByText } = render(<CodeSnippet {...makeProps()} />);
    expect(getByText('Copy')).toBeDefined();
  });

  it('should display the filename', () => {
    const { getByText } = render(<CodeSnippet {...makeProps()} />);
    expect(getByText('index.js')).toBeDefined();
  });

  it('should display the code', () => {
    const { getByText } = render(<CodeSnippet {...makeProps()} />);
    expect(getByText(/mainReducer/)).toBeDefined();
  });

  // TODO: right a valid test
  // it('should display Copied after clicking on the copy button', () => {
  //   const { getByText, debug } = render(<CodeSnippet {...makeProps()} />);
  //   fireEvent.click(getByText('Copy'));
  //
  //   // expect(getByText('Copied')).toBeDefined();
  // });
});
