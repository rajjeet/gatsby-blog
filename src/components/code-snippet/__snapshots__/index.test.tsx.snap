// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CodeSnippet /> should render 1`] = `
<DocumentFragment>
  .c4 {
  margin-right: .5em;
}

.c0 {
  background-color: rgba(33,133,208,.1);
  border-radius: 5px;
}

.c2 {
  font-family: Consolas,Monaco,'Andale Mono','Ubuntu Mono',monospace;
}

.c3 {
  background-color: white;
  border: none;
  padding: .5em 1em;
  border-radius: 5px;
}

.c1 {
  padding: .5rem .5rem 0 1rem;
  border-radius: 5px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <span
        class="c2"
      >
        index.js
      </span>
      <button
        class="c3"
        data-clipboard-text="
const mainReducer = (state, action) => {
  console.log('Before', state);
  console.log('Action', action);
  let result = {
    count: countReducer(state.count, action)      
  };
  console.log('After', result);
  return result
};      
"
        title=""
        type="button"
      >
        <span
          class="c4"
        >
          Copy
        </span>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-clipboard fa-w-12 "
          data-icon="clipboard"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M384 112v352c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48V112c0-26.51 21.49-48 48-48h80c0-35.29 28.71-64 64-64s64 28.71 64 64h80c26.51 0 48 21.49 48 48zM192 40c-13.255 0-24 10.745-24 24s10.745 24 24 24 24-10.745 24-24-10.745-24-24-24m96 114v-20a6 6 0 0 0-6-6H102a6 6 0 0 0-6 6v20a6 6 0 0 0 6 6h180a6 6 0 0 0 6-6z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
    <pre
      class=""
    >
      <code
        class="language-javascript"
      >
        
const mainReducer = (state, action) =&gt; {
  console.log('Before', state);
  console.log('Action', action);
  let result = {
    count: countReducer(state.count, action)      
  };
  console.log('After', result);
  return result
};      

      </code>
    </pre>
  </div>
</DocumentFragment>
`;
